### DISCUSSION TITLE
Python Testing Frameworks

### DISCUSSION THREAD

**Discussion by mikedorfman** [nsidc/tech-radar#10] [Languages and Frameworks]
Currently using Pytest as a test running framework. It can run unittest style tests.

Previously used nose/unittest.

**Comment by MattF-NSIDC** [comment]
lol we created the same card. How to avoid stepping on each others' toes like this?

https://github.com/nsidc/tech-radar/discussions/11

**Comment by bruceraup** [comment]
I think statements of action should have subjects, as in "*I* previously used nose/unittest.", so that we don't inadvertently speak for others.  I still use nosetests because it's faster than pytest in some cases (factor of 5ish? I haven't documented why).  Either one works with the tests I've written.

**Reply by MattF-NSIDC** [reply]
I think we discussed in the previous tech radar that Nose was deprecated because it's been in maintenance mode for ~8 years as of now. Nose2 exists, but is that actually being used by anyone here? It is a completely distinct project which does not support all of nose's features. https://docs.nose2.io/en/latest/differences.html

https://nose.readthedocs.io/en/latest/

> Nose has been in maintenance mode for the past several years and will likely cease without a new person/team to take over maintainership. New projects should consider using [Nose2](https://github.com/nose-devs/nose2), [py.test](http://pytest.org/), or just plain unittest/unittest2.

FWIW I've never observed a performance difference! May be worth opening a bug up with the pytest team if you can reproduce this?
